### ISSUE TITLE
Change Request: Error Messaging Updates

### ISSUE THREAD

**Issue by joellewells** [department-of-veterans-affairs/va-mobile-app#9627]
### 1.) Requestor's Name

Joelle Wells

### 2.) Requestor's Email

joelle.wells@oddball.io

### 3.) VA Team

Identity

### 4.) What is the name of the project or change?

MHV LOA1 Error Messaging Changes

### 5.) Please explain what problem you are trying to solve.

When users try to log in to the mobile app with a basic MHV LOA1 account, they receive a plain API error. It lacks clear guidance for users and front end styling. 

### 6.) Please explain how this change would work.

Mobile app should catch and display a user-friendly message that clearly explains what the user needs to do to resolve the error. It'll replace the API error message and improve user experience.

### 7.) Features Impacted

NA

### 8.) Please include all relevant documentation links for this change request.

Link to Laura's ticket: https://github.com/department-of-veterans-affairs/va.gov-team/issues/92628

### 9.) Does your team have all the necessary materials (documents, designs, processes, analytics, APIs, etc) in place for Mobile to make these changes right now?

Yes

### 10.) Has this change been discussed with VAHB POs (Ryan or Don)?

Yes

### 11.) Has this change been agreed upon by all impacted VA Stakeholders?

Yes

### 12.) What is your ideal due date for this change request?

As soon as is possible for the team

**Comment by laurwill** [comment]
Hi @joellewells @DonMcCaugheyUSDS @kellylein @Hallm13 @rtwell , 

Updating here to make sure the urgency of this ticket is clear — this incorrect error state is currently blocking all Veterans who sign in with LOA1 MHV accounts from accessing the mobile app. 

Happy to sync on this if that would be helpful!

**Comment by joellewells** [comment]
Hi @laurwill Thank you for adding this context! I didn't realize it was blocking users; I thought that blocking LOA1 accounts was expected behavior on the mobile app and that there's a workaround.

I'd be happy to meet to understand better if we could get @kellylein and/or others on the mobile team to join us for that conversation. I'd like to see what can be done and when they might be able prioritize it.

**Comment by laurwill** [comment]
Yes! And to clarify further: mobile app **does** intentionally block LOA1 accounts, but those users can access the mobile app by creating an LOA3 account and signing in with that instead. So we're blocking them in the sense that instead of telling them what steps to take to get access, we're showing them an API error with no information.

**Comment by DonMcCaugheyUSDS** [comment]
@joellewells @laurwill When I brought this to the mobile team, they asked me to double check where exactly this bare API error appears.

![Image](https://github.com/user-attachments/assets/693cec4b-b3bb-4bb3-9ce2-12e5dcb2cd30)

Looking at the image in the [linked #92628 ticket](https://github.com/department-of-veterans-affairs/va.gov-team/issues/92628), this appears to be happening in the embedded web sign in flow that the mobile app displays in a web view.  You can see this sign in flow in your browser here: https://staging.va.gov/sign-in/ .

Could you confirm exactly how you trigger this error on mobile?  If it happens in the embedded sign in web view, then it's not part of the mobile app code and needs to be looked at by the identity team.

**Comment by laurwill** [comment]
Hi @DonMcCaugheyUSDS , yes, it happens in the web flow, but @timwright12 suggested opening a mobile app change request (see [slack thread](https://dsva.slack.com/archives/CSFV4QTKN/p1726084246440689?thread_ts=1726059480.835179&cid=CSFV4QTKN)) since this will need cooperation between identity and mobile app to fix. Is there a better way to kick this off?

**Comment by DonMcCaugheyUSDS** [comment]
@laurwill Thanks for linking this together for me, and sorry I didn't connect it sooner.  Another question for you -- do you have any data on how frequently a veteran might hit this error condition?  This would help in prioritization.

@timwright12 Looking back at the slack thread linked above, I'm not sure what work there is for the mobile team to do here, aside from possibly some testing?  This seems like it would be 100% changes by the identity team, unless there's some web to mobile interaction that could be triggered.

**Comment by timwright12** [comment]
We were originally talking about displaying an error message inside the app, but if there isn't going to be an in-app update I agree that we don't need to be involved outside of helping with testing unless we want a eyes on the error message content.

The context from Slack:
> Essentially we would like to work with your team to display a better error message on the mobile app when a user tries to login with an MHV LOA1 account.

**Comment by laurwill** [comment]
Oh yes, let me clarify — the current, unhelpful error is in the web flow. The new, helpful error could appear inside the mobile app, or in the web flow, whichever is faster to implement. 

And we have content for this ready to go! Here's the new [standardized sign-in alert in VADS](https://design.va.gov/patterns/help-users-to/sign-in) for this situation (person signs in with LOA1 MHV account):
![Image](https://github.com/user-attachments/assets/951c5957-d46f-447c-a77c-1b58f079b21d)

And to answer the frequency question @DonMcCaugheyUSDS asked above, Joe said MHV reported 300K users with LOA1 MHV accounts, but we're not sure if they're all unique or whether they have other account types as well. Still, that's potentially up to 300k people who could be trying to access the mobile app and getting a deadend API error, when they should be able to switch account types and get access.

**Comment by joeniquette** [comment]
I still believe this change needs to be performed by the mobile app. The identity team produces the api error and then its up to the mobile team to handle this and display an error message. If the mobile team doesnt handle this then the subsequent reauth or account creation would direct the user to va.gov instead of through the VAHB auth flow. I could be missing something about how the VAHB app picks up errors so Im happy to discuss this more.

**Comment by BeccaTupaj** [comment]
@rtwell now that the MHV login option is not available, I believe this work is no longer needed. From my understanding, the error described in this ticket only happened when users logged in with an MHV account. Could you confirm? 
cc: @timwright12

**Comment by sterkenburgsara** [comment]
@BeccaTupaj correct!
